airflow/models/taskinstance.py

```diff
@@ -2361,6 +2361,8 @@ def _run_raw_task(
                     self.log.info(e)
                 if not test_mode:
                     self.refresh_from_db(lock_for_update=True, session=session)
+                _run_finished_callback(callbacks=self.task.on_skipped_callback, context=context)
+                session.commit()
```

Comment:
This should not be run in test mode, similar to other callbacks.

Reply:
Hmm. In other exception handling code we already commit even if test_mode. I wonder if it is a bug (and which)?

```
            except AirflowException as e:
                if not test_mode:
                    self.refresh_from_db(lock_for_update=True, session=session)
                # for case when task is marked as success/failed externally
                # or dagrun timed out and task is marked as skipped
                # current behavior doesn't hit the callbacks
                if self.state in State.finished:
                    self.clear_next_method_args()
                    session.merge(self)
                    session.commit()
                    return None
                else:
                    self.handle_failure(e, test_mode, context, session=session)
                    session.commit()
                    raise
            except (Exception, KeyboardInterrupt) as e:
                self.handle_failure(e, test_mode, context, session=session)
                session.commit()
                raise
```

Reply:
@uranusjr - any coments for that one, or can we just resolve it ?

Reply:
I see a lot of other approvals. Resolving this one - it always can come back here.